Orange/data/table/base.py

```diff
@@ -0,0 +1,1409 @@
+
+    def __iter__(self):
+        """Iterate over the rows of this TableBase as SeriesBase, breaking the pandas contract.
+
+        Returns
+        -------
+        generator
+            A generator of rows as SeriesBase objects.
+
+        Examples
+        --------
+            >>> iris = Table('iris')
+            >>> for row in iris.iloc[:5]:
+            >>>     print(row)
+
+        Notes
+        -----
+        This breaks the pandas contract! Pandas iterates over column names by default.
+        However, this only breaks __str__ and __repr__, which are reimplemented anyway.
+        """
+        for _, row in self.iterrows():
+            yield row
+
+    def __str__(self):
+        """Override the pandas representation to provide a more Orange-friendly one."""
+        max_displayed_rows = 30
+        sep = "\t\t"
+        sep_small = "\t"
+
+        table_over_limit = len(self) > max_displayed_rows
+        attrs = list(filter_visible(self.domain.attributes))
+        class_vars = list(filter_visible(self.domain.class_vars))
+        metas = list(filter_visible(self.domain.metas))
+        cols = attrs + class_vars + metas
+        roles = ["attr"] * len(attrs) + ["class"] * len(class_vars) + ["meta"] * len(metas)
+
+        if table_over_limit:
+            indices = list(range(0, max_displayed_rows // 2)) + \
+                      list(range(len(self) - (max_displayed_rows // 2), len(self)))
+            displayed_part = self.iloc[indices]
+            ellipsis_row = max_displayed_rows / 2 - 1
+        else:
+            indices = range(len(self))
+            displayed_part = self
+            ellipsis_row = None
+
+        # efficient string concatenation
+        with StringIO() as s:
+            # header, names
+            s.write("i" + sep_small + "idx" + sep_small)
+            for var in cols:
+                s.write(var.name + sep_small)
+            s.write("\n")
+
+            # header, types
+            s.write(sep)
+            for var in cols:
+                s.write(str(type(var))[:-(len("Variable"))].lower() + sep_small)
+            s.write("\n")
+
+            # header, roles
+            s.write(sep)
+            for role in roles:
+                s.write(role + sep_small)
+            s.write("\n")
+
+            # values
+            for i, (idx, row) in zip(indices, displayed_part.iterrows()):
+                s.write(str(i) + sep_small + str(idx) + sep_small)
+                for var in cols:
+                    s.write(var.repr_val(row[var]) + sep_small)
+                if i == ellipsis_row:
+                    s.write("\n" + (sep + "..." + sep) * (len(cols) // 2 + 1))
+                s.write("\n")
+
+            # footer, dtypes
+            s.write(sep)
+            for dtype in displayed_part.dtypes[cols]:
+                s.write(str(dtype) + sep_small)
+            s.write("\n")
+            s.seek(0)
+            return s.read()
+
+    def __repr__(self):
+        return str(self)
```

Comment:
Unless it [reconstructs the python object](https://docs.python.org/3/reference/datamodel.html#object.__repr__), don't need this either.